alert-common/src/main/java/com/synopsys/integration/alert/common/action/api/AbstractJobResourceActions.java

```diff
@@ -0,0 +1,211 @@
+        if (!hasPermissions) {
+            return new ActionResponse<>(HttpStatus.FORBIDDEN, AbstractResourceActions.FORBIDDEN_MESSAGE);
+        }
+        ValidationActionResponse validationResponse = validateResource(resource);
+        if (validationResponse.isError()) {
+            return new ActionResponse<>(validationResponse.getHttpStatus(), validationResponse.getMessage().orElse(null));
+        }
+        return createResource(resource);
+    }
+
+    @Override
+    public ActionResponse<List<JobFieldModel>> getAll() {
+        try {
+            Set<String> descriptorNames = descriptorAccessor.getRegisteredDescriptors()
+                                              .stream()
+                                              .filter(descriptor -> ALLOWED_JOB_DESCRIPTOR_TYPES.contains(descriptor.getType()))
+                                              .map(RegisteredDescriptorModel::getName)
+                                              .collect(Collectors.toSet());
+            if (!authorizationManager.anyReadPermission(List.of(ConfigContextEnum.DISTRIBUTION.name()), descriptorNames)) {
+                return new ActionResponse<>(HttpStatus.FORBIDDEN, AbstractResourceActions.FORBIDDEN_MESSAGE);
+            }
+            List<JobFieldModel> models = new LinkedList<>();
+            ActionResponse<List<JobFieldModel>> response = readAllResources();
+            List<JobFieldModel> allModels = response.getContent().orElse(List.of());
+            for (JobFieldModel jobModel : allModels) {
+                boolean includeJob = hasRequiredPermissions(jobModel.getFieldModels(), authorizationManager::hasReadPermission);
+                if (includeJob) {
+                    models.add(jobModel);
+                }
+            }
+            return new ActionResponse<>(HttpStatus.OK, models);
+        } catch (AlertException ex) {
+            return new ActionResponse<>(HttpStatus.INTERNAL_SERVER_ERROR, String.format("Error reading job configurations: %s", ex.getMessage()));
+        }
+
+    }
+
+    @Override
+    public ActionResponse<JobFieldModel> getOne(UUID id) {
+        if (null == id) {
+            return new ActionResponse<>(HttpStatus.BAD_REQUEST, AbstractResourceActions.RESOURCE_IDENTIFIER_MISSING);
+        }
+
+        Optional<JobFieldModel> optionalModel = findJobFieldModel(id);
+
+        if (optionalModel.isPresent()) {
+            JobFieldModel fieldModel = optionalModel.get();
+            boolean hasPermissions = hasRequiredPermissions(fieldModel.getFieldModels(), authorizationManager::hasReadPermission);
+            if (!hasPermissions) {
+                return new ActionResponse<>(HttpStatus.FORBIDDEN, AbstractResourceActions.FORBIDDEN_MESSAGE);
+            }
+            return new ActionResponse<>(HttpStatus.OK, fieldModel);
+        }
+
+        return new ActionResponse<>(HttpStatus.NOT_FOUND, null);
```

Comment:
When the constructor for `ActionResponse` would take a `null` parameter, we should either add a static initializer or a new constructor without the `message` parameter.

Reply:
We have a constructor that takes the status only.  I missed some places where that should be used.  I will fix that.